### PR TITLE
Limit block supports flag support to v2

### PR DESCRIPTION
**Pull request by youknowriad**
To avoid impacting existing dynamic blocks, this PRs only enable block supports server-side for blocks using apiVersion 2.

### CODE REVIEW

File: lib/block-supports/index.php [hunk 1]
```diff
@@ -37,7 +37,7 @@ function gutenberg_apply_block_supports( $block_content, $block ) {
 
 	$block_type = WP_Block_Type_Registry::get_instance()->get_registered( $block['blockName'] );
 	// If no render_callback, assume styles have been previously handled.
-	if ( ! $block_type || ! $block_type->render_callback ) {
+	if ( ! $block_type || ! $block_type->apiVersion || ! $block_type->apiVersion < 2 || ! $block_type->render_callback ) {
```
Comment (youknowriad) [inline lib/block-supports/index.php:40]: The `apiVersion` comes from block.json and is camel case (to match js rules), how did we solve this kind of issues in previous block.json properties cc @gziolo @mcsf @jorgefilipecosta

Comment (gziolo) [reply lib/block-supports/index.php:40]: We map everywhere to meet PHP coding style guidelines. It should be `api_version` in the block type object instance. I'm not sure about other places though.

Comment (youknowriad) [reply lib/block-supports/index.php:40]: So when we call `register_block_type` explicitly, we should pass `api_version` right?

Comment (gziolo) [reply lib/block-supports/index.php:40]: Correct, this is what `regiser_block_type_from_metadata` does for other settings:

https://github.com/WordPress/wordpress-develop/blob/3a19ff15f9013f7acf6951def3937807ea87733f/src/wp-includes/blocks.php#L197-L210

Comment (youknowriad) [reply lib/block-supports/index.php]: Ok so for now, I explicitly added `api_version` to the registration call. We should do a patch for Core to add the api_version to that array.

File: packages/block-library/src/latest-posts/edit.js [hunk 2]
```diff
@@ -522,6 +523,16 @@ class LatestPostsEdit extends Component {
 	}
 }
 
+function LatestPostBlockWrapper( props ) {
```
Comment (youknowriad) [inline packages/block-library/src/latest-posts/edit.js:526]: Ideally, we refactor the whole block to be a functional component, but this is better done on a separate PR. This should be enough for the moment.

File: lib/block-supports/index.php (hunk 1 above)
Comment (gziolo) [reply lib/block-supports/index.php]: Right, that works, too 👍

Comment (mcsf) [reply lib/block-supports/index.php]: I agree with translating to snake case in PHP and with subsequently patching `register_block_type_from_metadata`.

**Review by mcsf — APPROVED** [review]
This makes sense.

**Comment by youknowriad** [comment]
I think this one is not needed anymore since the API is explicit now.
